lib/pact_broker/client/hal/http_client.rb

```diff
@@ -127,6 +133,18 @@ def verbose?
           verbose || ENV["VERBOSE"] == "true"
         end
 
+        def custom_x509_certificate?
```

Comment:
Is this useful? Can we just use the presence of the env vars to work out whether to set the cert/key or not, the same way we do for the SSL cert env vars?